mlflow/pytorch/__init__.py

```diff
@@ -285,6 +287,108 @@ def gen_data():
+
+    :param state_dict: PyTorch model state dict to be saved.
+
+    :param artifact_path: Run-relative artifact path.
+
+    :param pickle_module: The module that PyTorch should use to serialize ("pickle") the specified
+                          ``pytorch_model``. This is passed as the ``pickle_module`` parameter
+                          to ``torch.save()``. By default, this module is also used to
+                          deserialize ("unpickle") the PyTorch model at load time.
+
+    :param kwargs: kwargs to pass to ``torch.save`` method.
+
+    .. code-block:: python
+        :caption: Example
+
+        import torch
+        import mlflow
+        import mlflow.pytorch
+        # X data
+        x_data = torch.Tensor([[1.0], [2.0], [3.0]])
+        # Y data with its expected value: labels
+        y_data = torch.Tensor([[2.0], [4.0], [6.0]])
+        # Partial Model example modified from Sung Kim
+        # https://github.com/hunkim/PyTorchZeroToAll
+        class Model(torch.nn.Module):
+            def __init__(self):
+               super().__init__()
+               self.linear = torch.nn.Linear(1, 1)  # One in and one out
+            def forward(self, x):
+                y_pred = self.linear(x)
+            return y_pred
+        # our model
+        model = Model()
+        criterion = torch.nn.MSELoss(size_average=False)
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+        # Training loop
+        for epoch in range(500):
+            # Forward pass: Compute predicted y by passing x to the model
+            y_pred = model(x_data)
+            # Compute and print loss
+            loss = criterion(y_pred, y_data)
+            print(epoch, loss.data.item())
+            #Zero gradients, perform a backward pass, and update the weights.
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+        # After training
+        for hv in [4.0, 5.0, 6.0]:
+            hour_var = torch.Tensor([[hv]])
+            y_pred = model(hour_var)
+            print("predict (after training)",  hv, model(hour_var).data[0][0])
+        # log the model
+        with mlflow.start_run() as run:
+            mlflow.log_param("epochs", 500)
+            mlflow.pytorch.log_state_dict(model, "models")
+
+            # logging scripted module
+            scripted_pytorch_model = torch.jit.script(model)
+            mlflow.pytorch.log_state_dict(scripted_pytorch_model, "models")
+    """
+    pickle_module = pickle_module or mlflow_pytorch_pickle_module
+
+    with TempDir() as tmp:
+        local_path = tmp.path("model")
+        save_state_dict(
+            state_dict=state_dict, path=local_path, pickle_module=pickle_module, **kwargs
+        )
+        mlflow.tracking.fluent.log_artifacts(local_path, artifact_path)
+
+
+@experimental
+def save_state_dict(state_dict, path, pickle_module=None, **kwargs):
+    """
+    Save the model as state dict to a path on the local file system
```

Comment:
Should we avoid using `model` in `xxx_state_dict` APIs because `model` indicates these APIs expect a model state dict but they actually don't?

Reply:
should we need to change the `load_state_dict` method as well - https://github.com/chauhang/mlflow/blob/6f85512eb34aa1886a5a2ab142d33477325672d8/mlflow/pytorch/__init__.py#L700 . The argument name is `model_uri`. It makes more sense to me to keep it as model uri.

Reply:
> It makes more sense to me to keep it as model uri.

Is this because it's rare for users to save a state_dict that doesn't contain a model?

Reply:
I have rephrased the save and log state dict methods (by removing `model`). In `load_state_dict`, the input to the method is an URI  and not the state dict itself. Hence, it makes more sense to me to keep it as such. If you feel otherwise, please let me know.